src/App.tsx

```diff
@@ -1,26 +1,224 @@
+import { Errors } from './types/Error';
+import { Filter } from './types/Filter';
+import { filterTodo } from './utils/helpers';
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<Errors | null>(null);
+  const [filter, setFilter] = useState<Filter>(Filter.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deleteTodoId, setDeleteTodoId] = useState<number | null>(null);
+  const [title, setTitle] = useState('');
+  const filteredTodo = filterTodo(todos, filter);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    const getData = async () => {
+      setIsLoading(true);
+      try {
+        const todosFromServer = await getTodos();
+
+        setTodos(todosFromServer);
+      } catch {
+        setError(Errors.Load);
+      }
+
+      setIsLoading(false);
+    };
+
+    getData();
+  }, []);
+
+  const areAllCompleted = todos.every(todo => todo.completed);
+
+  const addNewTodo = () => {
+    const titleTrimmed = title.trim();
+
+    if (!titleTrimmed) {
+      setError(Errors.EmptyTitle);
+
+      return;
+    }
+
+    const temp = {
+      title: titleTrimmed,
+      id: 0,
+      completed: false,
+      userId: USER_ID,
+    };
+
+    setTempTodo(temp);
+    setError(null);
+
+    postTodo(temp)
+      .then(res => {
+        setTodos(prev => [...prev, res]);
+        setTitle('');
+      })
+      .catch(() => setError(Errors.Add))
+      .finally(() => {
+        setTempTodo(null);
+      });
+  };
+
+  const deleteCurrentTodo = useCallback((id: number) => {
+    setDeleteTodoId(id);
+    deleteTodo(id)
+      .then(() => {
+        setTodos(prev => prev.filter(todo => todo.id !== id));
+      })
+      .catch(() => {
+        setError(Errors.Delete);
+      })
+      .finally(() => setDeleteTodoId(null));
+  }, []);
+
+  const clearCompleted = async () => {
+    const todoIdsToClear: number[] = [];
+
+    try {
+      for (const todo of todos) {
+        if (todo.completed) {
+          try {
+            await deleteTodo(todo.id);
+            todoIdsToClear.push(todo.id);
+          } catch {
+            setError(Errors.Delete);
+          }
+        }
+      }
+
+      setTodos(prevState =>
+        prevState.filter(todo => !todoIdsToClear.includes(todo.id)),
+      );
+    } catch {
+      setError(Errors.Delete);
+    }
+  };
+
+  const toggleAll = async () => {
+    try {
+      const isAnyUncompleted = todos.some(todo => !todo.completed);
+
+      const completedValue = isAnyUncompleted ? true : false;
+
+      await Promise.all(
+        todos.map(async todo => {
```

Comment:
Yep, something like that I have mentioned at my previous comment. However, to make it cleaner, let's firstly create the array with promises, just to separate this logic from Promise.all execution.